### PR TITLE
`d/azurerm_mysql_flexible_server` - New Datasource

### LINKED ISSUE #14971: Support for Data Source for azurerm_mysql_flexible_server
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support for a Data Source reference of azurerm_mysql_flexible_server would be helpful in retrieving back attributes such as the resource ID of an existing Azure MySQL Flexible Server.

### New or Affected Resource(s)

* azurerm_mysql_flexible_server

### Potential Terraform Configuration

```hcl
data "azurerm_mysql_flexible_server" "example" {
  name                = "existingMySqlServer"
  resource_group_name = "existingResGroup"
}

output "id" {
  value = data.azurerm_mysql_flexible_server.example.id
}
```

### PR DESCRIPTION
**Pull request by aristosvo**
Fixes #14971

Screenshot 2022-01-15 at 16 38 43

## ToDo
- [x] Docs

### CODE REVIEW

File: website/docs/d/mysql_flexible_server.html.markdown
Comment (favoretti) [inline website/docs/d/mysql_flexible_server.html.markdown:94]:
```suggestion
* `auto_grow_enabled` - Is Storage Auto Grow enabled?
```

File: website/docs/d/mysql_flexible_server.html.markdown
Comment (favoretti) [inline website/docs/d/mysql_flexible_server.html.markdown:86]:
```suggestion
* `start_hour` - The start hour of the maintenance window.
```

File: internal/services/mysql/mysql_flexible_server_data_source.go
Comment (favoretti) [inline internal/services/mysql/mysql_flexible_server_data_source.go:32]:
```suggestion
```
Lint is also not happy here :)

**Review by favoretti — CHANGES_REQUESTED** [review]

File: internal/services/mysql/mysql_flexible_server_data_source.go
Comment (aristosvo) [reply internal/services/mysql/mysql_flexible_server_data_source.go]: Huh? I'm confused it passed the checks.. Thanks!

File: website/docs/d/mysql_flexible_server.html.markdown
Comment (aristosvo) [reply website/docs/d/mysql_flexible_server.html.markdown]: Copy-paste of resource documentation, will add an update there as well! 👍🏽

File: website/docs/d/mysql_flexible_server.html.markdown
Comment (aristosvo) [reply website/docs/d/mysql_flexible_server.html.markdown]: 👍🏽

**Review by favoretti — APPROVED** [review]
LGTM, pending maintainers approval, thanks!

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for this @aristosvo

**Pull request merged**
